src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -169,12 +172,46 @@ private void handle(String consumeTag, Delivery delivery) {
         }
     }
 
+    private boolean resubscribe() {
+        LOGGER.info("Trying to resubscribe to queue.");
+
+        RabbitMQConfiguration configuration = this.connectionManager.get().getConfiguration();
+        int attemptsCount = 0;
+        int amountOfAttempts = configuration.getMaxRecoveryAttempts();
+        boolean successful = false;
+
+        while (attemptsCount < amountOfAttempts) {
```

Comment:
we should
1) register special monitors for readiness, liveness.
2) switch readiness to false.
3) try to unsubscribe some times.
4) switch liveness to false.
5) continue attempts to unsubscribe.
...
after success subscription 
unregister the monitors.